src/R/Components/Impl/Plots/Implementation/ViewModel/RPlotDeviceViewModel.cs

```diff
@@ -0,0 +1,279 @@
+
+            Debug.Assert(_device != null);
+            Debug.Assert(_device.ActivePlot != null);
+            await _plotManager.ExportToBitmapAsync(_device.ActivePlot, deviceName, outputFilePath, _lastPixelWidth, _lastPixelHeight, _lastResolution);
+        }
+
+        public async Task ExportToMetafileAsync(string outputFilePath) {
+            _shell.AssertIsOnMainThread();
+
+            Debug.Assert(_device != null);
+            Debug.Assert(_device.ActivePlot != null);
+            await _plotManager.ExportToMetafileAsync(_device.ActivePlot, outputFilePath, PixelsToInches(_lastPixelWidth), PixelsToInches(_lastPixelHeight), _lastResolution);
+        }
+
+        public async Task ExportToPdfAsync(string outputFilePath) {
+            _shell.AssertIsOnMainThread();
+
+            Debug.Assert(_device != null);
+            Debug.Assert(_device.ActivePlot != null);
+            await _plotManager.ExportToPdfAsync(_device.ActivePlot, outputFilePath, PixelsToInches(_lastPixelWidth), PixelsToInches(_lastPixelHeight));
+        }
+
+        public async Task RemoveActivePlotAsync() {
+            _shell.AssertIsOnMainThread();
+
+            Debug.Assert(_device != null);
+            Debug.Assert(_device.ActivePlot != null);
+            await _plotManager.RemovePlotAsync(_device.ActivePlot);
+        }
+
+        public async Task ClearAllPlotsAsync() {
+            _shell.AssertIsOnMainThread();
+
+            Debug.Assert(_device != null);
+            await _plotManager.RemoveAllPlotsAsync(_device);
+        }
+
+        public async Task NextPlotAsync() {
+            _shell.AssertIsOnMainThread();
+
+            Debug.Assert(_device != null);
+            await _plotManager.NextPlotAsync(_device);
+        }
+
+        public async Task PreviousPlotAsync() {
+            _shell.AssertIsOnMainThread();
+
+            Debug.Assert(_device != null);
+            await _plotManager.PreviousPlotAsync(_device);
+        }
+
+        public async Task ActivateDeviceAsync() {
+            _shell.AssertIsOnMainThread();
+
+            if (_device == null) {
+                await _plotManager.NewDeviceAsync(InstanceId);
+            } else {
+                await _plotManager.ActivateDeviceAsync(_device);
+            }
+        }
+
+        public void ClickPlot(int pixelX, int pixelY) {
+            _shell.AssertIsOnMainThread();
+
+            if (LocatorMode) {
+                var result = LocatorResult.CreateClicked(pixelX, pixelY);
+                EndLocatorMode(result);
+            }
+        }
+
+        public async Task CopyPlotFromAsync(Guid sourceDeviceId, Guid sourcePlotId, bool isMove) {
+            _shell.AssertIsOnMainThread();
+
+            if (_device == null) {
+                await _plotManager.NewDeviceAsync(InstanceId);
+            }
+
+            Debug.Assert(_device != null);
+            await _plotManager.CopyOrMovePlotFromAsync(sourceDeviceId, sourcePlotId, _device, isMove);
```

Comment:
It doesn't. As per our conversation, I'll make the commands that don't need to modify the view model go directly to plot manager instead.